server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -349,6 +351,97 @@ public Query rangeQuery(
                 context
             );
         }
+
+        public static class DateRangeDocValuesLoader extends BlockDocValuesReader.DocValuesBlockLoader {
+            private final String fieldName;
+
+            public DateRangeDocValuesLoader(String fieldName) {
+                this.fieldName = fieldName;
+            }
+
+            @Override
+            public Builder builder(BlockFactory factory, int expectedCount) {
+                return factory.longsFromDocValues(expectedCount);
+            }
+
+            @Override
+            public AllReader reader(LeafReaderContext context) throws IOException {
+                var docValues = context.reader().getBinaryDocValues(fieldName);
+                return new DateRangeDocValuesReader(docValues);
+                /*
```

Comment:
We probably shouldn't commit this without at least an explanation. We can just zap it for now.